runtime/Dart/lib/src/tree/src/tree.dart

```diff
@@ -68,15 +71,15 @@ abstract class ParseTree extends SyntaxTree {
   ///  minimal change, which is to add this method.
   ///
   ///  @since 4.7
-  set parent(RuleContext parent);
+  set parent(covariant ParseTree? parent); // Todo: review type fix
```

Comment:
Is there a reason to tighten the type?

Reply:
Getters and setters for a field shouldn't have different types (it is an error on 2.12). The getter for this field defined above returned a `ParseTree` value. We can either change the getter or the setter.